src/week1/calculator.go

```diff
@@ -0,0 +1,6 @@
+package main
+
+func eval(a, b float64, op func(a, b float64) (r float64, err error)) (r float64, err error) {
```

Comment:
we could define a type for this function `func(a, b float64) (r float64, err error)`.